File: docs/snippets/read_score_eth_key.rb
require 'eth'
require 'jwt'
require 'faraday'

key                  = Eth::Key.new(priv: ENV['ETH_PRIVATE_KEY'])
request              = JWT.decode(params[:request], nil, false).first['request']
request['signature'] = Eth::Utils.prefix_hex(key.personal_sign(Eth::Utils.hex_to_bin(request['hash'])))
authorization        = JWT.encode({ request: request }, nil, 'none')

response             = Faraday.get('https://rg.traity.com/score', nil, 'Authorization': "Bearer #{authorization}").body
score                = JSON.parse(response)

File: docs/snippets/read_score_eth_key.js
const axios = require('axios')
const jwt = require('jsonwebtoken')
const Accounts = require('web3-eth-accounts')

const account = (new Accounts()).privateKeyToAccount(process.env.ETH_PRIVATE_KEY)
const { request } = jwt.decode(req.query.request)
const signature = account.sign(request.hash)
const credentials = jwt.sign({ ...request, signature }, null, { algorithm: 'none' })
const scoreApiRes = await axios.get('https://rg.traity.com/score', {
  headers: { authorization: `Bearer ${credentials}` }
})
const score = scoreApiRes.data

